### DISCUSSION TITLE
BTT SFS 2.0 filament sensor connection

### DISCUSSION THREAD

**Discussion by madspeed** [mriscoc/Ender3V2S1#1078] [Q&A] [answered]
Can anyone help with firmware setup and wiring diagram to ender 3 v2 neo with 4.2.2 board?
The new BTT SFS 2.0 has two sensors - switch and motion, so 4 wires instead of 3. 
My firmware is 2.1.3

**Comment by mriscoc** [comment, marked as answer]
This filament sensor requires a custom build and is not currently supported.
Maybe you can use it as a BTT Smart filament motion sensor connecting the +V, GND, and Motion out pins to the runout sensor connector.

**Comment by madspeed** [comment]
What is the best filament sensor, which is supported?
Thank you.

**Reply by mriscoc** [reply]
If you already buy that sensor, maybe you can use it as a BTT Smart filament motion sensor which is currently supported.

**Reply by madspeed** [reply]
I will try to connect it to octoprint via rpi gpio contacts. This method allows to connect several filament sensors.
